src/main/java/com/darkona/adventurebackpack/handlers/PlayerEventHandler.java

```diff
@@ -113,17 +113,12 @@ public void onPlayerJump(LivingEvent.LivingJumpEvent event) {
 
     @SubscribeEvent
     public void pistonBootsUnequipped(LivingEvent.LivingUpdateEvent event) {
-        if (event.entityLiving instanceof EntityPlayer) {
+        if (event.entityLiving instanceof EntityPlayer && ConfigHandler.pistonBootsAutoStep) {
             EntityPlayer player = (EntityPlayer) event.entityLiving;
             if (Wearing.isWearingBoots(player)) {
-                if (!pistonBootsStepHeight) {
-                    pistonBootsStepHeight = true;
-                }
+                player.stepHeight = 1.001F;
             } else {
-                if (pistonBootsStepHeight) {
```

Comment:
Is this variable (`pistonBootsStepHeight`) necessary at all? If not, please remove it.